File: base/src/main/java/me/shetj/base/base/BaseSwipeBackActivity.java
package me.shetj.base.base;

import android.app.Activity;
import android.support.annotation.Keep;
import android.view.MotionEvent;

import com.aitangba.swipeback.SwipeBackHelper;

/**
 * @author shetj
 */
@Keep
public abstract class BaseSwipeBackActivity<T extends BasePresenter> extends BaseActivity<T> implements SwipeBackHelper.SlideBackManager {


    private SwipeBackHelper mSwipeBackHelper;

    @Override
    public boolean dispatchTouchEvent(MotionEvent ev) {
        if(mSwipeBackHelper == null) {
            mSwipeBackHelper = new SwipeBackHelper(this);
        }
        return mSwipeBackHelper.processTouchEvent(ev) || super.dispatchTouchEvent(ev);
    }

    @Override
    public Activity getSlideActivity() {
        return this;
    }

    @Override
    public boolean supportSlideBack() {
        return true;
    }

    @Override
    public boolean canBeSlideBack() {
        return true;
    }

    @Override
    public void finish() {
        if(mSwipeBackHelper != null) {
            mSwipeBackHelper.finishSwipeImmediately();
            mSwipeBackHelper = null;
        }
        super.finish();
    }
}

File: base/src/main/java/me/shetj/base/base/BaseSwipeBackActivity.kt
package me.shetj.base.base

import android.os.Bundle
import androidx.annotation.Keep
import android.view.View

import me.imid.swipebacklayout.lib.SwipeBackLayout
import me.imid.swipebacklayout.lib.Utils
import me.imid.swipebacklayout.lib.app.SwipeBackActivityBase
import me.imid.swipebacklayout.lib.app.SwipeBackActivityHelper

/**
 * @author shetj
 */
@Keep
abstract class BaseSwipeBackActivity<T : BasePresenter<*>> : BaseActivity<T>(), SwipeBackActivityBase {
    private var mHelper: SwipeBackActivityHelper? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mHelper = SwipeBackActivityHelper(this)
        mHelper!!.onActivityCreate()
    }

    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)
        mHelper!!.onPostCreate()
    }

    override fun <V : View> findViewById(id: Int): V? {
        val v = super.findViewById<V>(id)
        return if (v == null && mHelper != null) {
            mHelper!!.findViewById(id) as V
        } else v
    }

    override fun getSwipeBackLayout(): SwipeBackLayout {
        return mHelper!!.swipeBackLayout
    }

    override fun setSwipeBackEnable(enable: Boolean) {
        swipeBackLayout.setEnableGesture(enable)
    }

    override fun scrollToFinishActivity() {
        Utils.convertActivityToTranslucent(this)
        swipeBackLayout.scrollToFinishActivity()
    }
}